master/internal/api_workspace.go

```diff
@@ -446,6 +446,34 @@ func (a *apiServer) PostWorkspace(
 	}, nil
 }
 
+func (a *apiServer) deleteWorkspaceNamespaceBinding(ctx context.Context,
+	clusterNames []string, workspaceID int32, curUser *model.User,
+) error {
+	if err := workspace.AuthZProvider.Get().
+		CanSetWorkspaceNamespaceBindings(ctx, *curUser, workspaceID); err != nil {
+		return status.Error(codes.PermissionDenied, err.Error())
+	}
+	var namespaces []string
+	_, err := db.Bun().NewDelete().
+		Table("workspace_namespace_bindings").
+		Where("workspace_id = ?", workspaceID).
+		Where("cluster_name in (?)", bun.In(clusterNames)).
+		Returning("namespace").
+		Model(&namespaces).
+		Exec(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces {
+		err = a.m.rm.MaybeRemoveNamespace(ns)
```

Comment:
okay I see where MaybeRemoveNamespace is used. question still stands for multirm vs singls k8s